packages/web-components/src/components/masthead/__stories__/links.ts

```diff
@@ -397,73 +397,201 @@ const l1Data: LegacyMastheadL1 = {
+            heading: {
+              headingLevel: 2,
+              title: 'Heading level 2',
+              description:
+                'Description text lorem ipsum dolor sit amet commondo ligua eaget',
+            },
             items: [
               {
-                title: 'Product support',
-                url: 'https://www.ibm.com/analytics/support',
+                title: 'Link item',
+                url: 'https://example.com',
               },
               {
-                title: 'Expert gallery',
-                url: 'https://developer.ibm.com/technologies/blockchain/?lnk=hpmdev_dw&lnk2=learn',
+                title: 'Link item',
+                url: 'https://example.com',
+              },
+            ],
+          },
+          {
+            span: 1,
+            heading: {
+              headingLevel: 2,
+              title: 'Heading level 2',
+            },
+            items: [
+              {
+                title: 'Link item',
+                url: 'https://example.com',
+                description:
+                  'Description text lorem ipsum dolor sit amet commondo ligua eaget',
+              },
+              {
+                title: 'Link item',
+                url: 'https://example.com',
+                description:
+                  'Description text lorem ipsum dolor sit amet commondo ligua eaget',
+              },
+              {
+                title: 'Link item',
+                url: 'https://example.com',
+                description:
+                  'Description text lorem ipsum dolor sit amet commondo ligua eaget',
+              },
+              {
+                title: 'Link item',
+                url: 'https://example.com',
+                description:
+                  'Description text lorem ipsum dolor sit amet commondo ligua eaget',
+              },
+            ],
+          },
+          {
+            span: 1,
+            heading: {
+              headingLevel: 2,
+              title: 'Heading level 2',
+            },
+            items: [
+              {
+                title: 'Link item',
+                url: 'https://example.com',
+              },
+              {
+                title: 'Link item',
+                url: 'https://example.com',
+              },
+              {
+                title: 'Link item',
+                url: 'https://example.com',
+              },
+              {
+                title: 'Link item',
+                url: 'https://example.com',
+              },
+              {
+                title: 'Link item',
+                url: 'https://example.com',
               },
             ],
           },
         ],
       },
     },
+    {
+      title: 'Plain Link',
+      url: 'https://example.com',
+    },
   ],
+  actions: {
+    login: {
+      title: 'Log in',
+      url: 'https://example.com',
+    },
+    cta: {
+      title: 'Optional CTA',
+      url: 'https://example.com',
+    },
+  },
```

Comment:
Just more curiosity / edification. Does this structure get validated at some point? I see there are Typescript types for it, is Typescript validating it and going to throw errors at some point if it's off?

Reply:
@m4olivei @andy-blum There's already an `eslint` and `prettier` configured within the `ibm-dotcom-translations` (see - https://github.ibm.com/webstandards/ibm-dotcom-translations/blob/master/package.json#L14) that validates these JSON files and part of the existing CI checks.

Reply:
@proeung I was more asking from a data structure / data types / object shape POV.

Reply:
@m4olivei Like React's prop-types, right? The components don't do any of that sort of validation, no.

That can change, though! TypeScript can't help us here since we're already executing the compiled JS by the time we get data, right? But after some quick looking, I found there are some tools out there that can help us automatically generate validator functions based on our types. I wonder if this would have performance implications due to the sheer size of the masthead and our expected data payloads...

We could also go the route @andy-blum considered and validate the data generated by the translations repo script. This has the benefits of automatically verifying we aren't sending bad data from the content endpoint, but our test app that gets built as part of CI tooling already helps us verify the data is good, albeit manually.

In either case, this sounds like a great candidate for that list of improvements you've been adding to 🙂

Reply:
One thing to consider is that right now, we aren't generating L1 data, so it's on the consumers to shape their data correctly.

Reply:
> Like React's prop-types, right?

I wasn't thinking of that, but yeah something like that would be useful for consumers of the component. I was thinking alont the lines of validation in the translations repo like @andy-blum mentioned. Although, per @andy-blum 's last comment, I realise now that L1 data doesn't live in the translations repo.

> That can change, though! TypeScript can't help us here since we're already executing the compiled JS by the time we get data, right?

Right, makes sense.

> But after some quick looking, I found there are some tools out there that can help us automatically generate validator functions based on our types. I wonder if this would have performance implications due to the sheer size of the masthead and our expected data payloads.

The way React prop types work, they only get used in development and are stripped away in production, so the performance impact is mitigated. Would be interesting if this could be the case for our use case as well.

> We could also go the route @andy-blum considered and validate the data generated by the translations repo script. This has the benefits of automatically verifying we aren't sending bad data from the content endpoint.

That would be neat and give us some warm fuzzies about the output of the script. There should be a way to take our Typescript types and validate the generated translation files against them. Perhaps using [JSON schema](https://json-schema.org/) and a tool that [generates JSON schema from Typescript types](https://github.com/YousefED/typescript-json-schema).

> In either case, this sounds like a great candidate for that list of improvements you've been adding to slightly_smiling_face

Added :).